### DISCUSSION TITLE
Nginx and HTTPS redirects

### DISCUSSION THREAD

**Discussion by Veldrovive** [fastapi/fastapi#3779] [Questions] [answered]
I am using Nginx to put FastAPI behind an SSL reverse proxy. It all works well and I can `GET` content from my API. However, when I send a `POST` request, the server responds with a `location` header to an `HTTP` endpoint. I assume it is doing a rewrite to the `OPTIONS` request, but this is causing an error as it breaks the mixed content policy.

I have not had to work extensively with `HTTPS` in the past so I feel like I am slamming my head against a wall. Why is FastAPI rewriting this endpoint to an `HTTP` request instead of an `HTTPS` one when the original request was with `HTTPS`? Is it something I can change, or is this due to something underlying with the reverse proxy?

**Comment by ghandic** [comment]
Not sure about this - I haven't done too much with load balancing FastAPI. @Kludex any ideas?

**Comment by Veldrovive** [comment, marked as answer]
Whoops, forgot to close this.

The problem was that my frontend was requesting from a different domain than the backend was on. NGINX was then not setting the correct headers for FastAPI to respond with HTTPS redirects. The new version of [nginx-proxy](https://github.com/nginx-proxy/nginx-proxy) allows for virtual paths which allowed me to put the frontend and backend on the same domain.
